src/ranged.cpp

```diff
@@ -254,27 +275,33 @@ bool player::handle_gun_damage( item &it )
             }
         }
         // These are the dirtying/fouling mechanics
-        if( dirt < 10000 ) {
-            dirtadder = curammo_effects.count( "BLACKPOWDER" ) * ( 200 - ( firing->blackpowder_tolerance *
-                        2 ) );
-            if( dirtadder < 0 ) {
-                dirtadder = 0;
+        if( !curammo_effects.count( "NON-FOULING" ) && !it.has_flag( "NON-FOULING" ) ) {
+            if( dirt < 10000 ) {
+                dirtadder = curammo_effects.count( "BLACKPOWDER" ) * ( 200 - ( firing->blackpowder_tolerance *
+                            2 ) ); // this is the dirt-increasing number for shots fired with gunpowder-based ammo. Usually dirt level increases by 1, unless it's blackpowder, in which case it increases by a higher number, but there is a reduction for blackpowder resistance of a weapon.
+                if( dirtadder < 0 ) {
+                    dirtadder = 0;
+                }
+                if( curammo_effects.count( "BLACKPOWDER" ) ||
+                    dirt < 7150 ) { // in addition to increasing dirt level faster, regular gunpowder fouling is also capped at 7,150, not 10,000. So firing with regular gunpowder can never make the gun quite as bad as firing it with black gunpowder. At 7,150 the chance to jam is significantly lower (though still significant) than it is at 10,000, the absolute cap.
+                    it.set_var( "dirt", std::min( 10000, dirt + dirtadder + 1 ) );
```

Comment:
Probably for another PR, but you may want to consider moving max dirt into a constant, because as it's used here it's a magic number.

Reply:
I thought about doing this, but I think it would give the impression that you can just change the constant in the code and the system would continue to make sense. Actually lots of numbers only really work because it's 10,000, like the probability of misfire calculations.

Reply:
> only really work because it's 10,000, like the probability of misfire calculations.

If I understand the math correctly, you can calculate the misfire probability with `x_in_y( dirt_dbl * dirt_dbl * dirt_dbl, dirt_max_dbl * dirt_max_dbl * dirt_max_dbl )`, where `dirt_max_dbl` is the double value of the maximum dirt value, and the expression will continue to make sense even if `dirt_max_dbl` is changed.

Reply:
> > only really work because it's 10,000, like the probability of misfire calculations.
> 
> If I understand the math correctly, you can calculate the misfire probability with `x_in_y( dirt_dbl * dirt_dbl * dirt_dbl, dirt_max_dbl * dirt_max_dbl * dirt_max_dbl )`, where `dirt_max_dbl` is the double value of the maximum dirt value, and the expression will continue to make sense even if `dirt_max_dbl` is changed.

The max dirt level of 10,000 is only used in one chance calculation in ranged.cpp, and that's on line 257 when calculating the damage chance due to being unlubricated. The only other mentions of 10,000 are when checking to make sure the max dirt level is not exceeded. The misfire and high pressure damage chance are calculated using the CURRENT dirt level and a number that works, like 5555555555555 or 1000000000000. The max dirt level is not part of that formula.

Those numbers were chosen because they made sense to me at the time in terms of weighing jamming and damage more at the higher end of the dirt spectrum. Even the unlubricated damage value of 10,000 is mostly coincidental because I decided unlubed guns should be damaged at that sort of rate. You can easily choose 20,000 if you wanted to reduce the chance. I don't see any benefit in tying the number to some sort of universal max, though I suppose I could tie the two max (when it checks if an increase is to be allowed) checks to a constant. But changing the constant will mean unexpected consequences unless one goes into all the dirt-related checks to make sure the formulas still do what they want them to do at various dirt levels.

Reply:
> > only really work because it's 10,000, like the probability of misfire calculations.
> 
> If I understand the math correctly, you can calculate the misfire probability with `x_in_y( dirt_dbl * dirt_dbl * dirt_dbl, dirt_max_dbl * dirt_max_dbl * dirt_max_dbl )`, where `dirt_max_dbl` is the double value of the maximum dirt value, and the expression will continue to make sense even if `dirt_max_dbl` is changed.

Let me know if you still want it after I explained it below (the constant max).

Reply:
I thought the misfire probability was the third power of the ratio of fouling, that is, `r = dirt_dbl / 10000`. `p = r ^ 3`, where `r` is the ratio of fouling and `p` is the misfire probability. Similarly I thought the probability of gun damage was `1 / 5.555555555555 * p`. I don't know if this was your original intention, but I think it still makes sense with "weighing jamming and damage more at the higher end of the dirt spectrum", as the third power of a number does exactly that.

Reply:
> I thought the misfire probability was the third power of the ratio of fouling, that is, `r = dirt_dbl / 10000`. `p = r ^ 3`, where `r` is the ratio of fouling and `p` is the misfire probability. Similarly I thought the probability of gun damage was `1 / 5.555555555555 * p`. I don't know if this was your original intention, but I think it still makes sense with "weighing jamming and damage more at the higher end of the dirt spectrum", as the third power of a number does exactly that.

Well anyway I constanted it where intended and added notes.